wire-ios-sync-engine/Source/UserSession/UserSession.swift

```diff
@@ -0,0 +1,140 @@
+//
+// Wire
+// Copyright (C) 2023 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import WireDataModel
+
+/// An abstraction of the user session for use in the presentation
+/// layer.
+
+public protocol UserSession: AnyObject {
```

Comment:
maybe a stupid question, but why do we need an abstraction of ZMUserSession for the UI layer?

Reply:
`ZMUserSession` is a bit of a black hole as it sucks in every dependency that you can think of and is very hard to instantiate as a mock in testing. Lots of UI components often only need the session to perform a single task, so it's easier to have a lightweight mock. 

I think we need to better organize how the UI uses the session, but I would defer that for later.